client/src/components/Tool/utilities.js

```diff
@@ -0,0 +1,27 @@
+import $ from "jquery";
+import { getAppRoot } from "onload/loadConfig";
+import { copy } from "utils/clipboard";
+
+export function copyLink(toolId, message) {
+    copy(`${window.location.origin + getAppRoot()}root?tool_id=${toolId}`, message);
+}
+
+export function downloadTool(toolId) {
+    window.location.href = `${getAppRoot()}api/tools/${toolId}/download`;
+}
+
+export function openLink(url) {
+    window.open(url);
+}
+
+export function send(options, job_def) {
```

Comment:
This is the only reason jQuery is included and I'm not clear on why this function has to exist in the first place.

Here's something similar from Upload with just axios and native Form objects.

```js

import axios from "axios";
import { prependPath } from "utils/redirect";
import { UploadItem } from "../model";
import { History } from "components/History/model";

const api = axios.create({
    baseURL: prependPath("/api"),
});

export async function saveItemToHistory(opts = {}) {

    // ...bunch of other stuff omitted

    let response;
    try {
        const body = new FormData();
        body.set("tool_id", "upload1");
        body.set("history_id", history.id);
        body.set("inputs", JSON.stringify(toolInputs));

        response = await api.post("/tools", body);
    } catch (err) {
        throw new SaveItemError(item, history, err.message);
    }
    return response;
}


```

Reply:
@OlegZharkov pointed this code path out. After looking into the history of this, I noticed that this is a legacy code path which allows tools to submit the resulting form data to an arbitrary url. Since `data source` tools do not display the tool form, this path seems to be unnecessary, hence I removed it.